backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -211,6 +226,37 @@ public CommitDAORdbImpl(AuthService authService, RoleService roleService) {
               .setIdType(IDTypeEnum.IDType.VERSIONING_REPO_COMMIT_BLOB)
               .build(),
           datasetVersion.getTagsList());
+
+      long version = datasetVersion.getVersion();
+      if (commitPaginationDTO.getCommitEntities() != null
+          && !commitPaginationDTO.getCommitEntities().isEmpty()) {
+        CommitEntity parentEntity = commitPaginationDTO.getCommitEntities().get(0);
+        String parentCompositeId =
+            VersioningUtils.getVersioningCompositeId(
+                repositoryEntity.getId(), parentEntity.getCommit_hash(), location);
+        String parentVersion =
+            metadataDAO.getProperty(
+                session,
+                IdentificationType.newBuilder()
+                    .setIdType(IDTypeEnum.IDType.VERSIONING_REPO_COMMIT_BLOB)
+                    .setStringId(parentCompositeId)
+                    .build(),
+                ModelDBConstants.VERSION);
+        if (parentVersion != null && !parentVersion.isEmpty()) {
+          version = Long.parseLong(parentVersion) + 1L;
+        }
+      }
+      if (version == 0) {
+        version = version + 1;
```

Comment:
so version = 1 ??? 

we should only look at the version of parent commit or even fetch parent commit if there is no version set explicitly

```
if the request has a version save it , else increment the version of the parent commit and store it
```

Reply:
Change it according to comment 
```
if the request has a version save it, else increment the version of the parent commit and store it
```